cli/templatepull_test.go

```diff
@@ -230,118 +230,93 @@ func TestTemplatePull_LatestStdout(t *testing.T) {
+				_ = os.RemoveAll(tc.givenPath)
+				_ = os.RemoveAll(expectedDest)
+			})
 
-// ToDir tests that 'templates pull' pulls down the active template and writes
-// it to a directory with the name of the template if the path is not implicitly
-// supplied.
-// nolint: paralleltest
-func TestTemplatePull_ToImplicit(t *testing.T) {
-	client := coderdtest.New(t, &coderdtest.Options{
-		IncludeProvisionerDaemon: true,
-	})
-	owner := coderdtest.CreateFirstUser(t, client)
-	templateAdmin, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID, rbac.RoleTemplateAdmin())
+			client := coderdtest.New(t, &coderdtest.Options{
+				IncludeProvisionerDaemon: true,
+			})
+			owner := coderdtest.CreateFirstUser(t, client)
+			templateAdmin, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID, rbac.RoleTemplateAdmin())
 
-	// Create an initial template bundle.
-	source1 := genTemplateVersionSource()
-	// Create an updated template bundle. This will be used to ensure
-	// that templates are correctly returned in order from latest to oldest.
-	source2 := genTemplateVersionSource()
+			// Create an initial template bundle.
+			source1 := genTemplateVersionSource()
+			// Create an updated template bundle. This will be used to ensure
+			// that templates are correctly returned in order from latest to oldest.
+			source2 := genTemplateVersionSource()
 
-	expected, err := echo.Tar(source2)
-	require.NoError(t, err)
+			expected, err := echo.Tar(source2)
+			require.NoError(t, err)
 
-	version1 := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, source1)
-	_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, version1.ID)
+			version1 := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, source1)
+			_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, version1.ID)
 
-	template := coderdtest.CreateTemplate(t, client, owner.OrganizationID, version1.ID)
+			template := coderdtest.CreateTemplate(t, client, owner.OrganizationID, version1.ID)
 
-	// Update the template version so that we can assert that templates
-	// are being sorted correctly.
-	updatedVersion := coderdtest.UpdateTemplateVersion(t, client, owner.OrganizationID, source2, template.ID)
-	_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, updatedVersion.ID)
-	coderdtest.UpdateActiveTemplateVersion(t, client, template.ID, updatedVersion.ID)
+			// Update the template version so that we can assert that templates
+			// are being sorted correctly.
+			updatedVersion := coderdtest.UpdateTemplateVersion(t, client, owner.OrganizationID, source2, template.ID)
+			_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, updatedVersion.ID)
+			coderdtest.UpdateActiveTemplateVersion(t, client, template.ID, updatedVersion.ID)
 
-	// create a tempdir and change the working directory to it for the duration of the test (cannot run in parallel)
-	dir := t.TempDir()
-	wd, err := os.Getwd()
-	require.NoError(t, err)
-	err = os.Chdir(dir)
```

Comment:
How do these tests actually work with chdir removed? The files will be created inside the repo since we're not in the tmpdir, right?

Reply:
It pulls into the `givenPath`:

```go
inv, root := clitest.New(t, "templates", "pull", template.Name, tc.givenPath)
```

AFAICS it's functionally equivalent

Reply:
If the given path is relative, won't it be relative to the Go process, i.e. the test which is running in `coder/cli` directory. Thus it'll store files in there instead of tmp.

Reply:
You're correct! I added these two lines to account for that in the last commit:

```go
			newWD := t.TempDir()
			require.NoError(t, os.Chdir(newWD))
```

Reply:
You also need to restore the wd after the test has ended, otherwise it could break other tests.

Reply:
Oooh good point!